pkg/tracegen/templated.go

```diff
@@ -311,7 +317,13 @@ func (g *TemplatedGenerator) generateSpan(scopeSpans ptrace.ScopeSpans, tmpl *in
 		event.Attributes().EnsureCapacity(len(e.attributes) + len(e.randomAttributes))
 
 		event.SetName(e.name)
-		eventTime := start.Add(random.Duration(0, duration))
+
+		var eventTime time.Time
+		if e.percentageOfSpan > 0 {
+			eventTime = start.Add(time.Duration(float64(duration) * float64(e.percentageOfSpan)))
```

Comment:
I might be missing something here, but could you explain the rationale behind `e.percentageOfSpan` controlling the event timestamp?